src/main/java/object/GameObject.java

```diff
@@ -43,6 +44,53 @@ public void move() {
         body.shift(speed);
     }
 
-    public void intersectGameObject() {
+    @Override
+    public void takeDamage(int damage) {
+
+    }
+
+    @Override
+    public boolean isBroken() {
+        return false;
+    }
+
+    @Override
+    public boolean isDamageable() {
+        return false;
+    }
+
+    @Override
+    public int getDamage() {
+        return 0;
+    }
+
+    @Override
+    public boolean isProjectile() {
+        return false;
+    }
+
+    @Override
+    public Bonus getBonus() {
+        return null;
+    }
+
+    @Override
+    public boolean isBonus() {
```

Comment:
isBonus and isBonusTolerable. Whats the difference

Reply:
`isBonusTolerable()` - if bonus could be applied to a bonus. `isBonus()` - if gameObject is bonus